x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -501,18 +503,26 @@ protected LogicalPlan rule(UnaryPlan plan) {
         }
 
         private static LocalSupplier aggsFromEmpty(List<? extends NamedExpression> aggs) {
-            var result = new ArrayList<>(aggs.size());
+            Block[] blocks = new Block[aggs.size()];
+            var blockFactory = BlockFactory.getNonBreakingInstance();
+            int i = 0;
             for (var agg : aggs) {
                 // there needs to be an alias
                 if (agg instanceof Alias a && a.child() instanceof AggregateFunction aggFunc) {
-                    result.add(aggFunc instanceof Count ? 0L : null);
+                    Object value = aggFunc instanceof Count count && (count.foldable() == false || count.fold() != null) ? 0L : null;
```

Comment:
When does foldable == false and fold() != null ?

Reply:
The condition is looking for count(1) but not count(null) which should return null. Thus NOT, count(field)